rclcpp/include/rclcpp/experimental/subscription_intra_process_buffer.hpp

```diff
@@ -169,6 +170,72 @@ class SubscriptionIntraProcessBuffer : public SubscriptionROSMsgIntraProcessBuff
     return buffer_->use_take_shared_method();
   }
 
+  bool
+  is_serialized() const override
+  {
+    return serialization_traits::is_serialized_message_class<SubscribedType>::value;
+  }
+
+  SubscribedTypeUniquePtr deserialize_message(
+    const rclcpp::SerializedMessage * serialized_message)
+  {
+    if (serialized_message == nullptr) {
+      // The method is only allowed to be called with valid data (serialized case).
+      throw std::runtime_error("nullptr provided for serialized message");
+    }
+
+    auto ptr = SubscribedTypeAllocatorTraits::allocate(subscribed_type_allocator_, 1);
+    SubscribedTypeAllocatorTraits::construct(subscribed_type_allocator_, ptr);
+    if constexpr (std::is_same<SubscribedType, rclcpp::SerializedMessage>::value) {
+      (void)serialized_message;
+      throw std::runtime_error("Serialized message cannot be serialized.");
+    } else if constexpr (std::is_same<SubscribedType, ROSMessageType>::value) {
+      Serialization<SubscribedType> serializer;
+      serializer.deserialize_message(serialized_message, ptr);
+    } else {
+      ROSMessageType ros_msg;
+      Serialization<ROSMessageType> serializer;
+      serializer.deserialize_message(serialized_message, &ros_msg);
+      rclcpp::TypeAdapter<SubscribedType, ROSMessageType>::convert_to_custom(ros_msg, *ptr);
+    }
+
+    return SubscribedTypeUniquePtr(ptr, subscribed_type_deleter_);
+  }
+
+  bool serve_serialized_message(
+    const rclcpp::SerializedMessage * serialized_message,
+    IntraProcessManager * intraprocess_manager,
+    uint64_t intra_process_publisher_id,
+    void * untyped_allocator,
+    const std::vector<uint64_t> & take_ownership_subscriptions,
+    const std::vector<uint64_t> & take_shared_subscriptions) override
+  {
+    if (intraprocess_manager == nullptr) {
```

Comment:
do we need this check here? it looks like this variable is only forwarded to `do_intra_process_publish_same_type`